samples/remote_attestation/common/dispatcher.cpp

```diff
@@ -182,10 +182,174 @@ int ecall_dispatcher::verify_report_and_set_pubkey(
     return ret;
 }
 
+int ecall_dispatcher::establish_secure_channel(uint8_t** key, size_t* key_size)
+{
+    uint8_t encrypted_key_buf[512];
+    uint8_t signature[256];
+    uint8_t digest[32];
+    size_t encrypted_key_size = 256;
+    size_t signature_size = 256;
+    size_t total_size = 512;
+
+    int ret = 1;
+
+    if (m_initialized == false)
+    {
+        TRACE_ENCLAVE("ecall_dispatcher initialization failed.");
+        goto exit;
+    }
+
+    // Step 1 - Create an ephemeral symmetric key
+    if (oe_random(m_enclave_config->sym_key, 32) != OE_OK)
+    {
+        TRACE_ENCLAVE("ecall_dispatcher initialization failed to generate "
+                      "ephemeral symmetric key.");
+        goto exit;
+    }
+
+    TRACE_ENCLAVE("enclave: establish_secure_channel: Generated random "
+                  "symmetric key successfully");
+
+    // Step 2- Encrypt the symmetric key with the other enclave's public key
+    // TO DO - PREFIX with counter/seq number to prevent replay attacks
+    if (m_crypto->encrypt(
+            m_crypto->get_the_other_enclave_public_key(),
+            m_enclave_config->sym_key,
+            32,
+            encrypted_key_buf,
+            &encrypted_key_size))
+
+    {
+        // Reason we allocate memory in the host is so that the encrypted data
+        // can be accessed by the hosts - aka real world scenario where we have
+        // 2 hosts and two enclaves
+        uint8_t* host_buf = (uint8_t*)oe_host_malloc(total_size);
```

Comment:
I don't understand this comment. In Trust Zone, which is a real world scenario, enclaves cannot allocate memory on the host and then use the memory like a pointer.
@EMumau took a pass through even our test code and removed calls to oe_host_malloc (as much as possible) since it is bad programming practice.

Reply:
The existing sample still uses oe_host_malloc and so this would need to be removed (probably addressed by another PR).

Reply:
@anakrish - With regards to your earlier comment w.r.t. plan of action specified in #1707 , we (Cheng-mean or I) will need to update the PR with the reasoning for the change (if this gets in). Otherwise, further action will still need to be taken to fix the issues.

Reply:
You should at least replace the comment in 225 with a TODO saying this won't work and needs to be fixed.  However, since this is apparently new code, we shouldn't add (i.e., merge) new code that won't work.

Reply:
@dthaler - The existing samples do this and probably will not work for Trust Zone. Let's address this with a new issue. oe_host_malloc needs to be deprecated.